helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixDataAccessor.java

```diff
@@ -55,27 +54,19 @@
 public class ZKHelixDataAccessor implements HelixDataAccessor {
   private static Logger LOG = LoggerFactory.getLogger(ZKHelixDataAccessor.class);
   private final BaseDataAccessor<ZNRecord> _baseDataAccessor;
-  final InstanceType _instanceType;
   private final String _clusterName;
   private final Builder _propertyKeyBuilder;
   private final GroupCommit _groupCommit = new GroupCommit();
 
   public ZKHelixDataAccessor(String clusterName, BaseDataAccessor<ZNRecord> baseDataAccessor) {
-    this(clusterName, null, baseDataAccessor);
-  }
-
-  public ZKHelixDataAccessor(String clusterName, InstanceType instanceType,
```

Comment:
Although I agree with the change, this might break some users. Please note this PR as non-backward-compatible change.
And maybe we need to talk with the team.

Reply:
TFTR. Good point. How about keep the public constructor, mark it as deprecated, and remove this field.

Reply:
Given the meaning of this parameter is not possible to clarify, I actually support removing it.